go-controller/pkg/ovn/controller/unidling/unidle.go

```diff
@@ -56,7 +58,20 @@ func (uc *unidlingController) onServiceAdd(obj interface{}) {
 }
 
 func (uc *unidlingController) onServiceDelete(obj interface{}) {
-	svc := obj.(*v1.Service)
+	svc, ok := obj.(*v1.Service)
```

Comment:
Note to ourselves: it might be worth making a helper / util func for this, the same code applies to all our control loops and variants already exists in the `service_controller.go` and `factory/handler.go`

Reply:
Event handles in informers are not designed to have much logic so it is better to keep it this way, it is also how is implemented in Kubernetes upstream in most of the controllers so it keeps the code consistent and uniform